src/components/Comments/Comment.js

```diff
@@ -176,9 +194,27 @@ const Comment = ({
     },
     {
       id: 2,
-      name: loading.loadingDmAnon ? 'Loading...' : `Message ${username} anonymously`,
-      icon: <SendDMAnonBlock />,
-      onPress: onPressDMAnon
+      name:
+        loading.loadingGetAllowAnonDmStatus || loading.loadingDmAnon
+          ? 'Loading...'
+          : 'Message using Incognito Mode',
+      icon: (
+        <IcDmAnon
+          color={
+            !userAllowDm || loading.loadingGetAllowAnonDmStatus || loading.loadingDmAnon
+              ? COLORS.gray
+              : 'black'
+          }
+        />
+      ),
+      onPress: userAllowDm
+        ? onPressDMAnon
+        : () => {
+            Toast.show('This user does not allow messages in Incognito Mode.', Toast.SHORT);
+          },
+      style: (!userAllowDm || loading.loadingGetAllowAnonDmStatus || loading.loadingDmAnon) && {
+        color: COLORS.gray300
+      }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for displaying the 'Message using Incognito Mode' option has been updated. Now it checks if the user allows anonymous DMs or not. If not, it shows a toast message. Make sure the user experience is smooth and the toast message is clear to the user.

```javascript
{
  id: 2,
  name:
    loading.loadingGetAllowAnonDmStatus || loading.loadingDmAnon
      ? 'Loading...'
      : 'Message using Incognito Mode',
  icon: (
    <IcDmAnon
      color={
        !userAllowDm || loading.loadingGetAllowAnonDmStatus || loading.loadingDmAnon
          ? COLORS.gray
          : 'black'
      }
    />
  ),
  onPress: userAllowDm
    ? onPressDMAnon
    : () => {
        Toast.show('This user does not allow messages in Incognito Mode.', Toast.SHORT);
      },
  style: (!userAllowDm || loading.loadingGetAllowAnonDmStatus || loading.loadingDmAnon) && {
    color: COLORS.gray300
  }
}
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->